roofit/roofitcore/src/RooTreeDataStore.cxx

```diff
@@ -461,16 +461,23 @@ void RooTreeDataStore::loadValues(const TTree *t, const RooFormulaVar* select, c
   std::unique_ptr<TTree, decltype(deleter)> tClone(static_cast<TTree*>(t->Clone()), deleter);
   tClone->SetDirectory(t->GetDirectory());
 
-  // Clone list of variables  
+  // Clone list of variables
   std::unique_ptr<RooArgSet> sourceArgSet( _varsww.snapshot(kFALSE) );
-  
+
   // Check that we have the branches:
+  Bool_t missingBranches = kFALSE;
   for (const auto var : *sourceArgSet) {
-    if (!tClone->GetBranch(var->GetName())) {
-      coutE(InputArguments) << "Didn't find a branch in Tree '" << tClone->GetName()
-          << "' to read variable '" << var->GetName() << "' from."
-          << "\n\tNote: Name the RooFit variable the same as the branch." << std::endl;
-    }
+     if (!tClone->GetBranch(var->GetName())) {
+        missingBranches = kTRUE;
+        coutE(InputArguments) << "Didn't find a branch in Tree '" << tClone->GetName() << "' to read variable '"
+                              << var->GetName() << "' from."
+                              << "\n\tNote: Name the RooFit variable the same as the branch." << std::endl;
+     }
+  }
+  if (missingBranches) {
+     coutE(InputArguments) << "Cannot import ddata from TTree '" << tClone->GetName()
```

Comment:
```suggestion
     coutE(InputArguments) << "Cannot import data from TTree '" << tClone->GetName()
```

Reply:
Thanks Stephan for this ! I'll update the PR